api/src/room_store.py

```diff
@@ -60,9 +62,30 @@ def __init__(self, path):
     def get_all_room_ids(self) -> list:
         return list(self.stored_data.keys())
 
-    def write_room_data(self, room_id: str, data: Iterable[Token]) -> None:
+    async def write_room_data(self, room_id: str, data: Iterable[Token]) -> None:
         storable_data = list(map(asdict, data))
         self.stored_data[room_id] = storable_data
 
-    def read_room_data(self, room_id: str) -> Optional[dict]:
+    async def read_room_data(self, room_id: str) -> Optional[dict]:
         return self.stored_data.get(room_id)
+
+
+class DatabaseRoomStore:
+    def __init__(self, path):
+        self.path = path
+        self.db = None
+
+    @staticmethod
+    async def obtain(path):
```

Comment:
`connect`?

Reply:
I don't think connect makes sense as a factory function name. Maybe split out the connection part into another function, and the factory function creates the object, calls connect, and returns the object?

Reply:
sounds good